### ISSUE TITLE
Widget Javascript not detected.  It may not be installed or enabled properly.

### ISSUE THREAD

**Issue by nleclercq** [jupyter/help#131]
Hi all,

Whatever is the way is install ipywidgets I systematically get the following warning when I execute the 'first action' requiring ipywidgets to be loaded:  'Widget Javascript not detected.  It may not be installed or enabled properly'. It disappears if I reexecute the 'action'.

For production platforms, I use the ipywidgets-5 package delivered by miniconda 2 & 3. I have also have ipywidgets-6 beta6 on my dev platform. In both cases, I get the problem.  

Our users are a bit confused by this message. Is there a recipe it get rid of it?
Thanks.

**Comment by SylvainCorlay** [comment]
Are the widgets displaying despite of the error message?

**Comment by nleclercq** [comment]
Yes they are. 
To be honest, I would say that the problem is not systematic but the number of occurrence is really high.

**Comment by SylvainCorlay** [comment]
Does it only show when you try executing cells before the kernel is ready (right after you open the notebook?)

**Comment by nleclercq** [comment]
Right! If I let the kernel complete its startup sequence I don't get the warning message.
Here is a [live demo](https://www.dropbox.com/s/inip40rou3cda72/WidgetJSNotDetected.webm?dl=1) of the problem

**Comment by SylvainCorlay** [comment]
OK, this is a bug in notebook, which was introduced in https://github.com/jupyter/notebook/pull/1002 .

When one attempts to run cell while the startup sequence is not completed, we cache execute requests and send them all at once. However, this occurs before notebook extensions are loaded.

**Comment by nleclercq** [comment]
I see. So next release will fix that. Great! 
Thanks a lot Sylvain.

**Comment by draperjames** [comment]
@nleclercq are you still having the same problem? b/c I'm running the newest ipywidgets, jupyter notebook, on conda and the warning will still pop up randomly. Is there any way to suppress the the warning?

**Comment by draperjames** [comment]
@SylvainCorlay is there anyway to suppress this warning?

**Comment by Rutrus** [comment]
Same problem here. It seems to be all right in Jupyter but "javascript detection".
e.g. I'm using gmaps for Jupyter and It did't display the frame, only the warning.

**Comment by mivade** [comment]
I just encountered this issue and the only thing that made it go away was to restart the Jupyter server. It's especially annoying to see this output to the screen multiple times when using, for example, tqdm progress bars.

Would it be possible to have the warning only print once per notebook?
